[5.1.0] test_hpc_is_visible_tolerance failure

### Describe the bug

When running the CI tests while rebuilding the Debian package for version 5.1.0, I get the following test failure with both Python 3.11 and 3.12:
```
___________________________________________________________________________

    def test_hpc_is_visible_tolerance():
        hpc = Helioprojective(200*u.arcsec, 0*u.arcsec,
                              observer='earth', obstime='2023-08-03').make_3d()
    
        # Due to the limitations of numerical precision, the coordinate will be computed to be slightly
        # below the solar surface, and thus invisible when the tolerance is set to zero
>       assert not hpc.is_visible(tolerance=0*u.m)
E       assert not True
E        +  where True = <bound method Helioprojective.is_visible of <Helioprojective Coordinate (obstime=2023-08-03T00:00:00.000, rsun=695700....eliographicStonyhurst Coordinate for 'earth'>): (Tx, Ty, distance) in (arcsec, arcsec, AU)\n    (200., 0., 1.01025746)>>(tolerance=(0 * Unit("m")))
E        +    where <bound method Helioprojective.is_visible of <Helioprojective Coordinate (obstime=2023-08-03T00:00:00.000, rsun=695700....eliographicStonyhurst Coordinate for 'earth'>): (Tx, Ty, distance) in (arcsec, arcsec, AU)\n    (200., 0., 1.01025746)>> = <Helioprojective Coordinate (obstime=2023-08-03T00:00:00.000, rsun=695700.0 km, observer=<HeliographicStonyhurst Coordinate for 'earth'>): (Tx, Ty, distance) in (arcsec, arcsec, AU)\n    (200., 0., 1.01025746)>.is_visible
E        +    and   Unit("m") = u.m

sunpy/coordinates/tests/test_frames.py:264: AssertionError
```
From the comment, I would think that the test itself may be not really stable; maybe one should just remove the first asser?

### To Reproduce

_No response_

### Screenshots

_No response_

### System Details

- sunpy 5.10
- astropy 6.0.0
- Python 3.11/3.12
- all installed as Debian packages in Debian unstable

### Installation method

Linux package manager

--- a/changelog/7356.trivial.rst
+++ b/changelog/7356.trivial.rst
@@ -1,0 +1,1 @@
+Fixed an environment-specific failure of a unit test for :meth:`sunpy.coordinates.Helioprojective.is_visible`.

--- a/sunpy/coordinates/tests/test_frames.py
+++ b/sunpy/coordinates/tests/test_frames.py
@@ -259,9 +259,10 @@ def test_hpc_is_visible_tolerance():
     hpc = Helioprojective(200*u.arcsec, 0*u.arcsec,
                           observer='earth', obstime='2023-08-03').make_3d()
 
-    # Due to the limitations of numerical precision, the coordinate will be computed to be slightly
-    # below the solar surface, and thus invisible when the tolerance is set to zero
-    assert not hpc.is_visible(tolerance=0*u.m)
+    # Due to the limitations of numerical precision, the coordinate may be computed to be slightly
+    # below the solar surface, and thus may be invisible when the tolerance is set to zero
+    if hpc.is_visible(tolerance=0*u.m):
+        pytest.skip("Test already passes prior to increasing the tolerance.")
 
     assert hpc.is_visible(tolerance=1*u.m)
 

